fundingmanager.go

```diff
@@ -1087,28 +1075,70 @@ func newChanAnnouncement(localIdentity, remotePub *btcec.PublicKey,
 		FeeProportionalMillionths: 0,
 	}
 
-	return &chanAnnouncement{
-		chanAnn:    chanAnn,
-		edgeUpdate: chanUpdateAnn,
+	chanUpdateAnn.Signature, err = f.cfg.SignAnnouncement(chanUpdateAnn)
+	if err != nil {
+		return nil, errors.Errorf("unable to generate channel "+
+			"update announcement signature: %v", err)
 	}
+
+	// Channel proof should be announced in the separate message, so we
+	// already have the bitcoin signature but in order to construct
+	// the proof we also need node signature. Use message signer in order
+	// to sign the message with node private key.
+	nodeSig, err := f.cfg.SignAnnouncement(chanAnn)
+	if err != nil {
+		return nil, errors.Errorf("unable to generate node "+
+			"signature for channel announcement: %v", err)
+	}
+
+	bitcoinSig, err := f.cfg.SignNodeKey(localPubKey, localFundingKey)
+	if err != nil {
+		return nil, errors.Errorf("unable to generate bitcoin "+
+			"signature for node public key: %v", err)
+	}
+
+	proof := &lnwire.AnnounceSignatures{
+		ChannelID:        chanPoint,
+		ShortChannelID:   chanID,
+		NodeSignature:    nodeSig,
+		BitcoinSignature: bitcoinSig,
+	}
+
+	return &chanAnnouncement{
+		chanAnn:       chanAnn,
+		chanUpdateAnn: chanUpdateAnn,
+		chanProof:     proof,
+	}, nil
 }
 
 // announceChannel announces a newly created channel to the rest of the network
 // by crafting the two authenticated announcements required for the peers on the
 // network to recognize the legitimacy of the channel. The crafted
 // announcements are then send to the channel router to handle broadcasting to
 // the network during its next trickle.
-func (f *fundingManager) announceChannel(idKey, remoteIDKey *btcec.PublicKey,
-	channel *lnwallet.LightningChannel, chanID lnwire.ShortChannelID, localProof,
-	remoteProof *channelProof) {
-
-	// TODO(roasbeef): need a Signer.SignMessage method to finalize
-	// advertisements
-	chanAnnouncement := newChanAnnouncement(idKey, remoteIDKey, channel, chanID,
-		localProof, remoteProof)
+func (f *fundingManager) announceChannel(localIDKey, remoteIDKey *btcec.PublicKey,
+	localFundingKey, remoteFundingKey *btcec.PublicKey, chanID lnwire.ShortChannelID,
+	chanPoint wire.OutPoint) {
+
+	ann, err := f.newChanAnnouncement(
+		localIDKey,
+		remoteIDKey,
+		localFundingKey,
```

Comment:
Similar comment here about reverting the indentation/wrapping style that isn't used anywhere else within the codebase.